### PR TITLE
Add study on deploying VM with KubeVirt/Virtlet.

### PR DESCRIPTION
**Pull request by fmuyassarov**

### CODE REVIEW

File: vms-in-kubernetes/README.md [hunk 1]
```diff
@@ -218,7 +198,43 @@ spec:
         chpasswd: { expire: False }
 ```
 
-## Running virtual machines in Kubernetes with Virtlet
+### Requirements and installation
+
+There are some requirements towards the nodes that should have hardware
+acceleration and could support huge page support. For multiple interfaces on the
```
Comment (Xenwar) [inline vms-in-kubernetes/README.md]: should support 'huge page support'

File: vms-in-kubernetes/README.md [hunk 2]
```diff
@@ -218,7 +198,43 @@ spec:
...
+vm, Multus or Genie have to be deployed on the cluster. Otherwise a CNI is
```
Comment (Xenwar) [inline vms-in-kubernetes/README.md]: VM or vm, use them consistently.

File: vms-in-kubernetes/README.md [hunk 3]
```diff
@@ -319,6 +443,29 @@ are far less than in KubeVirt, for example there is no live migration.
 Virtlet is much more complex than KubeVirt, requires more additional components
```
Comment (Xenwar) [inline vms-in-kubernetes/README.md]: requires additional components

File: vms-in-kubernetes/README.md [hunk 4]
```diff
@@ -319,6 +443,29 @@ are far less than in KubeVirt, for example there is no live migration.
...
 and seems to receive far less attention from the community.
 
+Furthermore, performance comparision has been conducted and according to
+results, KubeVirt reports better results on interface bandwidth (measured with ```iPerf```) but
+closely the same on other system resources (e.g. CPU Throlling, IdleJitter,
+etc) which is measured with [Netdata](https://github.com/netdata/netdata).\
+Please note that resaults of CPU performance below don't give insights on real CPU
+performance but rather explain the behaviour of various systems components when
```
Comment (Xenwar) [inline vms-in-kubernetes/README.md]: various system components

File: vms-in-kubernetes/README.md [hunk 5]
```diff
@@ -319,6 +443,29 @@ are far less than in KubeVirt, for example there is no live migration.
...
+imposed with certain load. As virtual machine doesn't support some of the
```
Comment (Xenwar) [inline vms-in-kubernetes/README.md]: which virtual machine ?

File: vms-in-kubernetes/README.md [hunk 6]
```diff
@@ -319,6 +443,29 @@ are far less than in KubeVirt, for example there is no live migration.
...
+standard CPU performance counters (CPU instructions, cycles...) it is
+problematic to measure real CPU performance on virtualized environment. As such,
```
Comment (Xenwar) [inline vms-in-kubernetes/README.md]: on a virtualized environment

File: vms-in-kubernetes/README.md [hunk 7]
```diff
@@ -319,6 +443,29 @@ are far less than in KubeVirt, for example there is no live migration.
...
+the CPU comparision below gives some idea on all the system components
```
Comment (Xenwar) [inline vms-in-kubernetes/README.md]: system components'

File: vms-in-kubernetes/README.md [hunk 8]
```diff
@@ -1,4 +1,4 @@
-# Running vms in a Kubernetes cluster
+# Running VMs in a Kubernetes cluster
 
 Two main options are available : KubeVirt and Virtlet. The first runs the
```
[the remark this answers is not in the document]
Comment (fmuyassarov) [reply vms-in-kubernetes/README.md]: resolved

File: vms-in-kubernetes/README.md (hunk 1 above)
Comment (fmuyassarov) [reply vms-in-kubernetes/README.md]: resolved

File: vms-in-kubernetes/README.md (hunk 2 above)
Comment (fmuyassarov) [reply vms-in-kubernetes/README.md]: resolved

File: vms-in-kubernetes/README.md (hunk 3 above)
Comment (fmuyassarov) [reply vms-in-kubernetes/README.md]: resolved

File: vms-in-kubernetes/README.md (hunk 4 above)
Comment (fmuyassarov) [reply vms-in-kubernetes/README.md]: resolved

File: vms-in-kubernetes/README.md (hunk 5 above)
Comment (fmuyassarov) [reply vms-in-kubernetes/README.md]: resolved

File: vms-in-kubernetes/README.md (hunk 7 above)
Comment (fmuyassarov) [reply vms-in-kubernetes/README.md]: resolved

File: vms-in-kubernetes/README.md (hunk 1 above)
Comment (Xenwar) [reply vms-in-kubernetes/README.md]: support huge pages

Comment (fmuyassarov) [reply vms-in-kubernetes/README.md]: done

File: vms-in-kubernetes/README.md [hunk 9]
```diff
@@ -316,9 +440,35 @@ Definitions. Virtlet implements pods with Libvirt, hence is more restricted in
...
-Virtlet is much more complex than KubeVirt, requires more additional components
+Virtlet is much more complex than KubeVirt, requires additional components
 and seems to receive far less attention from the community.
 
+Furthermore, performance comparision has been conducted and according to
+results, KubeVirt reports higher interface bandwidth (measured with
+```iPerf```) but
+closely the same on other system resources (e.g. CPU Throlling, IdleJitter,
+etc) which was measured with [Netdata](https://github.com/netdata/netdata).\
+Please note that resaults of CPU performance below don't give insights on real
```
Comment (maelk) [inline vms-in-kubernetes/README.md]: the results ( +the -a in resaults)

File: vms-in-kubernetes/README.md [hunk 10]
```diff
@@ -316,9 +440,35 @@ Definitions. Virtlet implements pods with Libvirt, hence is more restricted in
...
+
+```bash
+# generates load on 4 CPUs
+$ stress --cpu 4 --io 4 --vm 4 --vm-bytes 1280M --timeout 300s
+```
+
+|| Virtlet | KubeVirt |
+| ------------- | ------------- | ------------- |
+| Average bandwidth | 1.05 Gbits/sec | 821 Mbits/sec |
+| CPU metrics | [Virtlet](https://github.com/Nordix/airship-clusterapi-docs/blob/study/feature-feruz/vms-in-kubernetes/performance_veirtlet.pdf) | [Kubevirt](https://github.com/Nordix/airship-clusterapi-docs/blob/study/feature-feruz/vms-in-kubernetes/performance_kubevirt.pdf) |
```
Comment (maelk) [inline vms-in-kubernetes/README.md]: veirtlet typo

Comment (fmuyassarov) [reply vms-in-kubernetes/README.md]: resolved

File: vms-in-kubernetes/README.md (hunk 9 above)
Comment (fmuyassarov) [reply vms-in-kubernetes/README.md]: resolved

**Pull request merged**